app/controllers/admin/categories_controller.rb

```diff
@@ -1,27 +1,56 @@
 class Admin::CategoriesController < Admin::BaseController
   before_action :load_cate, except: %i(new index create)
-  def index; end
+  def index
+    @q = Category.ransack params[:q]
+    @categories = @q.result(distinct: true)
+      .show_cate.select_cate_desc.includes(:tours)
```

Comment:
sử dụng includes để giảm số lượng truy vấn xuống còn 1,  SELECT "tours".* FROM "tours" WHERE "tours"."category_id" IN (?, ?, ?, ?, ?), nếu không dùng thì số lượng truy vấn e là N+1 , do đã phân trang hiện tại page chỉ có 5 truy vấn trên 1 trang.